### ISSUE TITLE
slight modifications so it works on Windows `

### ISSUE THREAD

**Issue by gedw99** [sapcc/go-makefile-maker#76]
Just wanted to see if there is any interest in supporting windows ?

for example
``` 
DESTDIR =
ifeq ($(shell uname -s),Darwin)
	PREFIX = /usr/local
else
	PREFIX = /usr
endif
````

There are a few other places like where awk and grep are calls.
There are golang equivalents of these out there that i have been using and so can help contrib.

Then the CI can run matrix on windows also.

**Comment by gedw99** [comment]
AWK replacement :

https://github.com/benhoyt/goawk

**Comment by SuperSandro2000** [comment]
> Just wanted to see if there is any interest in supporting windows ?

No, generally not. Our workflow on Windows is to use WSL or ssh into a linux VM.

> There are golang equivalents of these out there that i have been using and so can help contrib.

We are using the gnu or bsd variants locally and there are most likely some minor differences, which we don't need to work around.

**Comment by majewsky** [comment]
To add to what Sandro said, our main challenge in adding nativ eWindows support would be to keep it working in the future, since we don't have any systems to test on. Also, and this is potentially a naive question from my side, how do Makefiles work on Windows anyway? Windows does not come with a `make`. The only way I've seen `make` run on Windows (besides WSL or VMs) is something like Cygwin, at which point you have a standard Unix userland anyway.

**Comment by gedw99** [comment]
@majewsky 
Make files work perfectly on windows.

Here is an example:
https://github.com/amplify-edge/booty

It means your CI calls a makefile.  Same dependencies on laptop and in CI.
here is an example: https://github.com/amplify-edge/booty/blob/master/.github/workflows/ci.yml#L50
- runs on all desktops and servers fine.

The advantage is that a dev can run the makefile and he CI will run exactly the same thing. so we don't get the old works on my laptop !! §&$%$"  

Booty is called by other repos make and hence their CI to install dependencies. so now their CI works and the exact same thing happens on a devs laptop. Its turtles all the way down...
https://github.com/amplify-edge/booty/blob/master/README.md#installation

**Comment by majewsky** [comment]
Ok, in that case I want to understand the scope better. Based on the `$(shell ...)` invocations that we currently have?

* Is `uname` available on Windows? If not, how do we do OS detection?
* Is `date` available on Windows? If not, what's the alternative?
* For all of these alternatives (including goawk instead of awk), how do we switch to them? Presumably via an OS detection thing.

What I would certainly consider to be out of scope is having code in our generate Makefiles for installing coreutils replacements like goawk. As said above, we have no interest in specific testing coverage for Windows, so this code would just be dead weight to us.

**Comment by gedw99** [comment]
i am not on windows. am on Mac. 

https://github.com/amplify-edge/main/blob/master/Makefile#L2

```
ifeq ($(OS),Windows_NT)
	BOOTY_URL:=$(BOOTY_URL)/install.ps1
else
	BOOTY_URL:=$(BOOTY_URL)/install.sh
endif
```
